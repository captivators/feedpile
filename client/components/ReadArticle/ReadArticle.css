.read-article-page {
  height: 100vh;
  background-color: #f2f2f2;
}

.read-article-container {
  width: 60%;
  max-width: 680px;
  margin: auto;
  font-family: "Roboto", sans-serif;
  color: #333;
  line-height: 150%;
}

.read-article-title {
  text-align: center;
  font-size: 28px;
  line-height: 120%;
  font-weight: bold;
  margin-bottom: 25px;
}

.default-image {
  text-align: center;
  margin-bottom: 25px;
}

.read-article-content img {
  max-width: 100%;
  height: auto;
  margin-bottom: 25px;
}

.url-button-container {
  margin-top: 20px;
  text-align: center;
  margin-bottom: 50px;
}

.url-button {
  display: inline-block;
}
