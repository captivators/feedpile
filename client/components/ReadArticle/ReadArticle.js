import React from 'react';
import { connect } from 'react-redux';
import './ReadArticle.css';

const ReadArticle = (props) => {
  return (
    <div className="reader-list-container">
      {props.articles.map((article, index) => (
        <ReaderListItem article={article} key={index}/>
      ))}
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    articles: state.articles
  }
};

export const Unwrapped  = ReadArticle;
export default connect(mapStateToProps)(ReadArticle);
