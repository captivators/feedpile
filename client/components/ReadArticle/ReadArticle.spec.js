import React from 'react';
import {Unwrapped as UnwrappedReadArticle} from './ReadArticle';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const article = {
  author: 'victoriawlau',
  date: '2017-04-17T21:04:38.000Z',
  datePublished: '2017-04-17T21:04:38.000Z',
  description: 'Test',
  feedId:'58f4faea29be910cb6467df0',
  imageSrc:'http://tctechcrunch2011.files.wordpress.com/2017/04/3-education-20-20.jpg',
  summary:'We asked the office and the TechCrunch moms on staff for things they’d like for↵Mother’s Day this year. While the unanimous answer to what they wanted was “a↵nap” – they weren’t opposed to receiving these gifts either. Read More',
  title:"12 tech-inspired Mother’s Day gifts",
  url:'http://feedproxy.google.com/~r/Techcrunch/~3/BsZCJrFPZRA/',
  _id:'58f52dfe619dacc31152c9d2'
}

test('ReadArticle take a snapshot', () => {
  const component = shallow(<UnwrappedReadArticle selectedArticle={article}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
