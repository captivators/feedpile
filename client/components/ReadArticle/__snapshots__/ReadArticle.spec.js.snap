// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadArticle take a snapshot 1`] = `
<MuiThemeProvider>
  <div
    className="read-article-page"
  >
    <Link
      replace={false}
      to="/app"
    >
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={
          Object {
            "height": 36,
            "width": 36,
          }
        }
        style={
          Object {
            "height": 72,
            "padding": 16,
            "width": 72,
          }
        }
        tooltipPosition="bottom-center"
        touch={false}
      >
        <pure(Close) />
      </IconButton>
    </Link>
    <div
      className="read-article-container"
    >
      <div
        className="read-article-title"
      >
        12 tech-inspired Motherâ€™s Day gifts
      </div>
      <div
        className="read-article-content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Test",
          }
        }
      />
    </div>
    <div
      className="url-button-container"
    >
      <RaisedButton
        disabled={false}
        fullWidth={false}
        href="http://feedproxy.google.com/~r/Techcrunch/~3/BsZCJrFPZRA/"
        icon={<pure(OpenInNew) />}
        label="Visit Website"
        labelPosition="after"
        primary={false}
        secondary={false}
        style={
          Object {
            "margin": 12,
          }
        }
        target="_blank"
      />
    </div>
  </div>
</MuiThemeProvider>
`;
